user_profiles/serializers.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib.auth import get_user_model
+
+from rest_framework import serializers
+
+from .models import UserProfile
+
+
+User = get_user_model()
+
+
+class ControlPKField(serializers.PrimaryKeyRelatedField):
+
+    def get_queryset(self):
+        user = self.context['request'].user
+        queryset = user.profile.controls.all()
+        return queryset
+
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    controls = ControlPKField(many=True)
+    first_name = serializers.CharField(source='user.first_name')
+    last_name = serializers.CharField(source='user.last_name')
+    email = serializers.EmailField(source='user.email')
+
+    class Meta:
+        model = UserProfile
+        fields = ('first_name', 'last_name', 'email', 'profile_type', 'organization', 'controls')
+        extra_kwargs = {'controls': {'write_only': True}}
+
+    def create(self, validated_data):
+        profile_data = validated_data
+        controls_data = profile_data.pop('controls')
```

Comment:
Pour la lisibilité : (j'aime bien améliorer la lisibilité, je trouve ca important, on pourra en discuter)
Je bougerais cette ligne plus bas, juste avant `for control_data ...` 
Comme ca on fait d'abord tout le `user_data`, ensuite on rajoute le `control_data`. C'est plus clair à lire.

Reply:
Ok, on change après les merges.